fst/en/inflection.py

```diff
@@ -0,0 +1,100 @@
+"""
+Inflects English nouns - creates a plural form from singular.
+
+Install Pynini package before running.
+"""
+
+import pynini as p
+
+from pynini.lib import pynutil
+from pynini.lib import rewrite
+
+def _priority_union(q: p.Fst, r: p.Fst, sigma: p.Fst) -> p.Fst:
+  complement_domain_q = sigma - p.project(q, 'input')
+  return p.union(q, complement_domain_q @ r)
+
+_v = p.union('a', 'e', 'i', 'o', 'u')
+_c = p.union('b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n',
+             'p', 'q', 'r', 's', 't', 'v', 'w', 'x', 'y', 'z')
+_sigma = p.union(_v, _c).closure().optimize()
+
+_exceptions = p.string_map([
+    # Zero plurals.
+    'deer',
+    'fish',
+    'sheep',
+    'series',
+    'species',
+    # Stem changes.
+    ('child', 'children'),
+    ('goose', 'geese'),
+    ('man', 'men'),
+    ('woman', 'women'),
+    ('tooth', 'teeth'),
+    ('foot', 'feet'),
+    ('mouse', 'mice'),
+    ('person', 'people'),
+    ('penny', 'pence'),
+    # Irregular suffixes.
+    ('child', 'children'),
+    ('ox', 'oxen'),
+    # f -> v.
+    ('wife', 'wives'),
+    ('wolf', 'wolves'),
+    # A few Greek and Latin plurals.
+    ('analysis', 'analyses'),
+    ('ellipsis', 'ellipses'),
+    # More exceptions
+    ('bus', 'busses'),
+    ('fez', 'fezzes'),
+    ('photo', 'photos'),
+    ('piano', 'pianos'),
+    ('halo', 'halos'),
```

Comment:
This list of exceptions for the "o" ending is much larger than the regular rule below. I recommend inverting this part of the logic.